src/com/github/igotyou/FactoryMod/interactionManager/FurnCraftChestInteractionManager.java

```diff
@@ -21,21 +29,137 @@
 public class FurnCraftChestInteractionManager implements IInteractionManager {
 	private FurnCraftChestFactory fccf;
 	private HashMap<Clickable, InputRecipe> recipes = new HashMap<Clickable, InputRecipe>();
+	private ReinforcementManager rm;
 
 	public FurnCraftChestInteractionManager(FurnCraftChestFactory fccf) {
 		this.fccf = fccf;
+		prepCitadel();
 	}
 
 	public FurnCraftChestInteractionManager() {
+		prepCitadel();
 	}
 
 	public void setFactory(FurnCraftChestFactory fccf) {
 		this.fccf = fccf;
 	}
+	
+	private void prepCitadel() {
+		if (FactoryModPlugin.getManager().isCitadelEnabled()) {
+			rm = Citadel.getReinforcementManager();
+		} else {
+			rm = null;
+		}
+	}
 
 	public void redStoneEvent(BlockRedstoneEvent e) {
-		// TODO
+		if (rm != null) {
+			// Note this also accomplishes all the Citadel checking we need.
+			BlockFace powerFace = findPoweringFace(e.getBlock(), fccf.getMultiBlockStructure().getAllBlocks());
+			if (powerFace != null) {
+				int trueNewCurrent = e.getBlock().getBlockPower(powerFace);
+				if (trueNewCurrent != e.getNewCurrent()) {
+					e.setNewCurrent(trueNewCurrent);
+				}
+			} else { // null means citadel is enabled but no valid redstone power was found.
+				return;
+			}
+		}
+
+		if (e.getNewCurrent() == e.getOldCurrent()) {
+			return;
+		}
+		int threshold = FactoryModPlugin.getManager().getRedstonePowerOn();
+		boolean newState = false;
+		if (e.getBlock().getLocation().equals(fccf.getFurnace().getLocation())) {
+			if (e.getOldCurrent() >= threshold && e.getNewCurrent() < threshold && fccf.isActive()) {
+				// Falling Edge (turn off)
+				newState = false;
+			} else if (e.getOldCurrent() < threshold && e.getNewCurrent() >= threshold && !fccf.isActive()) {
+				// Rising Edge (turn on)
+				newState = true;
+			} else {
+				return;
+			}
+			
+			if (newState) {
+				fccf.activate();
```

Comment:
Use this method instead to activate it https://github.com/Civcraft/FactoryMod/blob/FactoryMod2-0/src/com/github/igotyou/FactoryMod/factories/FurnCraftChestFactory.java#L72